TOMP-API.yaml

```diff
@@ -1681,162 +1521,264 @@ components:
+        - endTime
       properties:
         from:
-          description: The coordinates the TO should use to resolve leg start location
+          description: The coordinate the TO should use to resolve leg start location
           $ref: '#/components/schemas/place'
         to:
-          description: The coordinates the TO should use to resolve leg finish location
+          description: The coordinate the TO should use to resolve leg finish location
           $ref: '#/components/schemas/place'
-        start-time:
+        startTime:
           $ref: '#/components/schemas/timestamp'
-        end-time:
+        endTime:
           $ref: '#/components/schemas/timestamp'
         mode:
-          $ref: '#/components/schemas/type-of-asset'
+          $ref: '#/components/schemas/typeOfAsset'
         state:
-          $ref: '#/components/schemas/leg-state'           
-        departure-delay:
+          $ref: '#/components/schemas/legState'
+        departureDelay:
           $ref: '#/components/schemas/duration'
-        arrival-delay:
+        arrivalDelay:
           $ref: '#/components/schemas/duration'
         distance:
           $ref: '#/components/schemas/distance'
         fare:
           $ref: '#/components/schemas/fare'
         route:
           type: string
-        route-short-name:
+        routeShortName:
+          type: string
+        routeLongName:
+          type: string
+        agencyId:
           type: string
-        route-long-name:
+        legGeometry:
           type: string
-        agency-id:
+          description: format as in geojson linestring eg. [[6.169639, 52.253279], .. ] WGS84, [lng,lat]
+        assetAccessData:
+          $ref: '#/components/schemas/token'
+          description: data to open a specific asset (e.g. QR code, image base64). This can be provided when using assign_asset, when it's not provided in the booking.
+
+    legEvent:
+      type: object
+      description: event for the leg
+      required:
+        - time
+        - event
+      properties:
+        time:
+          $ref: '#/components/schemas/timestamp'
+        event:
           type: string
-        leg-geometry:
+          enum: [PREPARE, ASSIGN_ASSET, SET_IN_USE, PAUSE, START_FINISHING, FINISH, ISSUE]
+        comment:
           type: string
-          description: format as in geojson linestring eg. [[6.169639, 52.253279], .. ] WGS84, [lng,lat]
-        asset-access-data:
-          $ref: '#/components/schemas/token'
-          description: data to open a specific asset (e.g. QR code, image base64). This can be provided when using assign_asset, when it's not provided in the booking.
-          
-    leg-progress:
+          description: free text
+        asset:
+          $ref: '#/components/schemas/asset'
+
+    legProgress:
       type: object
       description: provides current asset location & duration and distance of the current leg
       required:
-        - coordinates
+        - coordinate
```

Comment:
coordinate is a req. field, but in the properties below it's still coordinate**s**

Reply:
Check, I'll repair this one